app/src/main/java/com/weatherxm/usecases/RewardsUseCase.kt

```diff
@@ -19,9 +31,17 @@ interface RewardsUseCase {
         deviceId: String,
         date: ZonedDateTime
     ): Either<Failure, RewardDetails>
+
+    suspend fun getBoostReward(
+        deviceId: String,
+        boostReward: BoostReward
+    ): Either<Failure, UIBoost>
 }
 
-class RewardsUseCaseImpl(private val repository: RewardsRepository) : RewardsUseCase {
+class RewardsUseCaseImpl(
```

Comment:
Am not sure if the UI's Resources class should be used inside a UseCase class. It's a transformation to create a "State UI" model that I would consider moving to ViewModel. Or you can create a model without Resource IDs and in ViewModel class you can add all the contexted-based staff.

Reply:
I remember when I had some architecture-related discussions back then, that one functionality of a UseCase would be to prepare the UI Model for the upper layers (ViewModel) and the ViewModel to prepare / have available anything else it might need to the activity.

That's why I defined this model here, what do you think of that? Shall we create the UI Model in the ViewModel?

Btw we have used this pattern in other UseCases as well afaik, but of course if we decide to change this pattern we can optimize it in others as well (in the future).